public/tabs/chat/chat_page_content.test.tsx

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import { ChatPageContent } from './chat_page_content';
+import * as chatContextExports from '../../contexts/chat_context';
+import * as chatStateHookExports from '../../hooks/use_chat_state';
+import * as chatActionHookExports from '../../hooks/use_chat_actions';
+import { IMessage } from '../../../common/types/chat_saved_object_attributes';
+
+jest.mock('./messages/message_bubble', () => {
+  return {
+    MessageBubble: ({ children }: { children?: React.ReactNode }) => (
+      <div aria-label="chat message bubble">{children}</div>
+    ),
+  };
+});
+
+jest.mock('./messages/message_content', () => {
+  return { MessageContent: () => <div /> };
+});
```

Comment:
just curious why are child components mocked?

Reply:
That's a good question! There has been lots of discussions around this. For **unit test**, I tend to mock child components in RTL which are smart components, for example, the components that:
1. Access data via API call
2. Have side effects such as read/write local storage
3. Access context
4. Access global state

By mocking such components, I think it makes the tests concise and less fragile. And ofc, the child component should be covered by its own unit tests.